packages/styled-base/src/utils.js

```diff
@@ -1,9 +1,31 @@
 // @flow
 import * as React from 'react'
+import type { ElementType } from 'react'
 import isPropValid from '@emotion/is-prop-valid'
 
 export type Interpolations = Array<any>
 
+export type StyledOptions = {
+  label?: string,
+  shouldForwardProp?: string => boolean,
+  target?: string
+}
+
+export type StyledComponent = {
```

Comment:
I'm not sure this is the best way to define this type, but I'm unable to extend the built-in component types.

Reply:
yeah - had the same issue when I've tried to handle this once I've found the culprit, couldnt just add `withComponent` to `StatelessFunctionComponents` and others